### ISSUE TITLE
How to catch client side authentication failures

### ISSUE THREAD

**Issue by jaimewyant** [Code-Sharp/WampSharp#214]
I am using dynamic ticket authentication to authenticate users.  When authentication fails, the authentication component raises an ApplicationError.  How should this be handled in my WampSharp code?

This is what the python code looks like ([full example here](https://github.com/crossbario/crossbar-examples/blob/master/authentication/advanced/authenticator.py))

```python
            if ticket != principal['ticket']:
               raise ApplicationError(u'com.example.invalid_ticket', "could not authenticate session - invalid ticket '{}' for principal {}".format
```

I expected an exception while awaiting channel.Open(), but it just doesn't return.

Am I overlooking something?

**Comment by darkl** [comment]
A WampConnectionBrokenException should be raised with the relevant properties.

**Comment by darkl** [comment]
Ok, I checked your example. Here's the thing: crossbario doesn't seem to the close the underlying connection in your case. On the other hand, WampSharp expects the underlying connection to be closed, and only then it fires the exception.
This should probably be fixed. Will try to find time for that.

Elad

**Comment by darkl** [comment]
I committed a fix, but my CI is currently broken for some reason. Hopefully
it will fixed in the few next days, and then I could release a new version
with the fix.

Elad

**Comment by jaimewyant** [comment]
Thanks for the quick turn around Elad.  I was going to take a stab at it myself, but c# isn't my "native language".

Thanks again!

**Comment by darkl** [comment]
I just released WampSharp 1.2.6.42-beta to NuGet gallery. It contains a fix for this problem.
Elad

**Comment by jaimewyant** [comment]
Worked perfectly for me.  Thanks!
